beacon_node/network/src/service.rs

```diff
@@ -733,12 +734,13 @@ impl<T: BeaconChainTypes> NetworkService<T> {
                 }
 
                 if !self.subscribe_all_subnets {
-                    for column_subnet in
-                        DataColumnSubnetId::compute_subnets_for_data_column::<T::EthSpec>(
-                            self.network_globals.local_enr().node_id().raw().into(),
-                            &self.beacon_chain.spec,
-                        )
-                    {
+                    for column_subnet in DataColumnSubnetId::compute_custody_subnets::<T::EthSpec>(
+                        self.network_globals.local_enr().node_id().raw().into(),
+                        self.network_globals
+                            .local_enr()
+                            .custody_subnet_count::<<T as BeaconChainTypes>::EthSpec>()
+                            .unwrap_or(self.beacon_chain.spec.custody_requirement),
```

Comment:
This custody_subnet_count value will need to be available inside the BeaconChain to decide when `PendingComponents` are available. This value is likely to not change at runtime so there can be multiple sources of truth, but just wondering what's the best source for it.

Reply:
I've been looking into an related issue and trying to get the `custody_subnets` into beacon chain, however the tricky part is `network_global.local_enr` is initialised as part of the networking stack after the `beacon_chain` is constructed, so we need some way of making that value available in DA checker.

Reply:
Had a good discussion with Lion and Age on this, and I think we've found a good path forward regarding the above comment - We might need to move things around later, but i think the code here is fine as it is for now!